uaclient/apt.py

```diff
@@ -560,3 +566,23 @@ def setup_unauthenticated_repo(
         apt_origin,
         "never",
     )
+
+
+def compare_versions(version1: str, version2: str, relation: str) -> bool:
+    """Return True comparing version1 to version2 with the given relation."""
+    try:
+        system.subp(
+            ["dpkg", "--compare-versions", version1, relation, version2]
+        )
+        return True
+    except exceptions.ProcessExecutionError:
+        return False
+
+
+def get_apt_cache_time():
+    cache_time = inf
```

Comment:
This probably deserves a docstring explaining why `inf` is the default return value

Alternatively, this function could not assume what the caller needs, and just return `None` if it can't find the time. Then the caller can determine if that should be treated as "inf" or 0 or an error